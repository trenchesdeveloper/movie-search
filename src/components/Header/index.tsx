import React from "react";
import { Link } from "react-router-dom";

import RMDBLogo from "../../images/react-movie-logo.svg";
import OMDBLogo from "../../images/tmdb_logo.svg";

import { Wrapper, Content, LogoImg, OMDBLogoImg } from "./Header.styles";

const Header: React.FC = () => {
  console.log("header", "render");

  return (
    <Wrapper>
      <Content>
        <Link to="/">
          <LogoImg src={RMDBLogo} alt="rmdb-logo" />
        </Link>
        <OMDBLogoImg src={OMDBLogo} alt="tmdb-logo" />
      </Content>
    </Wrapper>
  );
};

export default Header;
