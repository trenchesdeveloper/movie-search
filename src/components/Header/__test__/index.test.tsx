import { render, screen } from "@testing-library/react";
import Header from "..";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { BrowserRouter } from "react-router-dom";

const MockHeader = () => {
  return (
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
};
describe("Header", () => {
  test("Header component renders correctly a header", () => {
    render(<MockHeader />);
    const header = screen.getByRole("heading");

    expect(header).toBeInTheDocument();
  });

  test("Header component renders the two logo images", async () => {
    render(<MockHeader />);
    const logo = await screen.findAllByRole("img");
    expect(logo).toHaveLength(2);
  });

  test("Header component snapshot", () => {
    const tree = renderer.create(<MockHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
