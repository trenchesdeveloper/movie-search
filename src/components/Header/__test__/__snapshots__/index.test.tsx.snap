// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header component snapshot 1`] = `
.c0 {
  background: var(--darkGrey);
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: var(--maxWidth);
  padding: 20px 0;
  margin: 0 auto;
}

.c2 {
  width: 200px;
}

.c3 {
  width: 100px;
}

@media screen and (max-width:500px) {
  .c2 {
    width: 150px;
  }
}

@media screen and (max-width:500px) {
  .c3 {
    width: 80px;
  }
}

<h1
  className="c0"
>
  <div
    className="c1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="rmdb-logo"
        className="c2"
        src="react-movie-logo.svg"
      />
    </a>
    <img
      alt="tmdb-logo"
      className="c3"
      src="tmdb_logo.svg"
    />
  </div>
</h1>
`;
